desktop/main.cpp

```diff
@@ -0,0 +1,282 @@
+    m_liveReload = liveReload;
+    Q_EMIT liveReloadChanged();
+  }
+  QUrl codeLocation() const { return m_codeLocation; }
+  void setCodeLocation(const QUrl &codeLocation) {
+    if (m_codeLocation == codeLocation)
+      return;
+    m_codeLocation = codeLocation;
+    Q_EMIT codeLocationChanged();
+  }
+  QString pluginsPath() const { return m_pluginsPath; }
+  void setPluginsPath(const QString &pluginsPath) {
+    if (m_pluginsPath == pluginsPath)
+      return;
+    m_pluginsPath = pluginsPath;
+    Q_EMIT pluginsPathChanged();
+  }
+  QString executor() const { return m_executor; }
+  void setExecutor(const QString &executor) {
+    if (m_executor == executor)
+      return;
+    m_executor = executor;
+    Q_EMIT executorChanged();
+  }
+  QString packagerHost() const { return m_packagerHost; }
+  void setPackagerHost(const QString &packagerHost) {
+    if (m_packagerHost == packagerHost)
+      return;
+    m_packagerHost = packagerHost;
+    setCodeLocation(m_packagerTemplate.arg(m_packagerHost).arg(m_packagerPort));
+  }
+  QString packagerPort() const { return m_packagerPort; }
+  void setPackagerPort(const QString &packagerPort) {
+    if (m_packagerPort == packagerPort)
+      return;
+    m_packagerPort = packagerPort;
+    setCodeLocation(m_packagerTemplate.arg(m_packagerHost).arg(m_packagerPort));
+  }
+  void setLocalSource(const QString &source) {
+    if (m_localSource == source)
+      return;
+
+    // overrides packager*
+    if (source.startsWith("file:")) {
+      setCodeLocation(source);
+    } else {
+      QFileInfo fi(source);
+      if (!fi.exists()) {
+        qWarning() << "Attempt to set non-existent local source file";
+        return;
+      }
+      setCodeLocation(QUrl::fromLocalFile(fi.absoluteFilePath()));
+      setLiveReload(false);
+    }
+  }
+Q_SIGNALS:
+  void liveReloadChanged();
+  void codeLocationChanged();
+  void pluginsPathChanged();
+  void executorChanged();
+
+private:
+  bool m_liveReload = false;
+  QString m_packagerHost = "localhost";
+  QString m_packagerPort = "8081";
+  QString m_localSource;
+  QString m_packagerTemplate =
+      "http://%1:%2/index.desktop.bundle?platform=desktop&dev=true";
+  QUrl m_codeLocation;
+  QString m_pluginsPath;
+#ifdef BULID_FOR_BUNDLE
+  QString m_executor = "RemoteServerConnection";
+#else
+  QString m_executor = "LocalServerConnection";
+#endif
+};
+
+#ifdef BULID_FOR_BUNDLE
+void runUbuntuServer();
+void saveMessage(QtMsgType type, const QMessageLogContext &context,
+                 const QString &msg);
+
+void writeLogsToFile();
+#endif
+
+int main(int argc, char **argv) {
+
+  QGuiApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
+  QGuiApplication app(argc, argv);
+
+  Q_INIT_RESOURCE(react_resources);
+
+#ifdef BULID_FOR_BUNDLE
+  QString dataFolder = QDir::homePath() + "/Library/StatusIm/";
+  qInstallMessageHandler(saveMessage);
+
+  QDir dir(dataFolder + "ethereum/mainnet_rpc");
+  if (!dir.exists()) {
+    dir.mkpath(".");
```

Comment:
data folder creation here, probably, is not required, if to apply changes to rctstatus.cpp mentioned below.
P.S. Default Network type to place geth node on, seems, is defined in .env file and passed to rctstatus::startNode finally.